### DISCUSSION TITLE
[Question]:  custom name for GPT isn't working?

### DISCUSSION THREAD

**Discussion by TheValkyrja** [danny-avila/LibreChat#2541] [Q&A]
### What is your question?

I have found that the custom name setting for GPT does not reflect the correct context in the system prompt. I'm not sure if it is a bug or if it is designed that the change in the custom name for GPT only changes its display name.

### More Details

I tried to set a custom name for GPT on the website, but the setting only changes the display name for GPT, instead of sending a prompt reference to the assistant's name for it in the system role.
In comparison, when setting a custom name for Claude 3 family, the backend sends a "You are xxx" in the system instruction.
i wonder if it is a bug or if it just should work that way.

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/45366459/9453c5ec-a399-4206-b6e6-5f363c13b580)
![image](https://github.com/danny-avila/LibreChat/assets/45366459/5d56f864-c423-4d02-8127-bf861c2ba90b)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It depends on the API. Yes, for Anthropic, there is a bit of a custom prompt to better steer the LLM to recognize its name, because it may not have an explicit field for a name (I don't remember off the top of my head).

For OpenAI, messages have a `name` field. It can be seen here in the docs:
https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages

This is how these messages look when sent to the API
![image](https://github.com/danny-avila/LibreChat/assets/110412045/43f41c66-9dbd-4cb2-a6cd-e51b23b5e5a9)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/96d935c2-60c3-428c-ba16-8b4da78df764)


and yeah it looks like the API isn't giving much weight to the field as I thought it would for a specific name, so we can add it to the instructions to improve this.

**Reply by TheValkyrja** [reply]
I've read the api reference before i post the question. Since the name field isn't necessary, personally I tend to believe it's more like only a tag for differenciate participants of same role.
for example, in playground you could set up two consecutive contexts, both with the user role.If the two contexts are not from the same "speaker", they can be distinguished by name while using same role.
I think referencing the assistant's name again in the system instruction is a more reliable option.

**Reply by danny-avila** [reply]
sounds good, I agree, I can add this easily
